R/tar_insuvimeh.R

```diff
@@ -0,0 +1,123 @@
+catalogue_insuvimeh_files <-  function(gdb){
+  fps <- list.files(gdb,
+                    full.names =T,
+                    recursive = T,
+                    pattern = "\\d{4}.nc$")
+  fps_w_parents <- list.files(gdb,
+                              recursive = T,
+                              pattern = "\\d{4}.nc$")
+  lr <- fps %>% 
+    map2(fps_w_parents,\(fp_nc,fp_par){
+      
+      fp_base <- basename(fp_nc)
+      # meta data collection  - to write band_name as "{pub_date}.lt_{leadtime}"
+      valid_time_meta <- ncmeta::nc_att(fp_nc,"T","units")$value$units
+      start_time_meta <- ncmeta::nc_att(fp_nc,"S","units")$value$units
+      valid_date <- as_date(str_extract(valid_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      start_date <- as_date(str_extract(start_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      lead_time_diff <- interval(start_date,valid_date)
+      lt <- lead_time_diff %/% months(1)
+      tibble(
+        file_name_parent = fp_par,
+        file_name= fp_base,
+        pub_date = start_date,
+        valid_date =valid_date,
+        lt = lt
+      )
+    }) %>% 
+    list_rbind()
+  return(lr)
+}
+
+
+load_insuvimeh_raster <-  function(gdb){
+  fps <- list.files(gdb,
+                    full.names =T,
+                    recursive = T,
+                    pattern = "\\d{4}.nc$")
+  
+  lr <- fps %>% 
+    map(\(fp_nc){
```

Comment:
Again, I think having well documented/commented functions rather than really long anonymous ones would be better practice.

Reply:
yeah good point. Def should add some roxygen and comments. Perhaps I can break it up a bit too. 

Your point is still relevant, but just so you know: filenames associated w/ custom functions specific to this targets pipeline project (like this one) all start with `tar_*.R`. In target work flows it's encouraged to write these custom functions to keep the pipeline/intent more svelt, but they aren't necessarily written with being generic in mind.

Reply:
agree looks cleaner w/ your suggestion. Added more documentation roxygen too [5f422a6](https://github.com/OCHA-DAP/ds-aa-lac-dry-corridor/pull/12/commits/5f422a6a41ad97a3497dc7a4702cb7defc9890ae)